pkg/controller/deployment/deployment_controller.go

```diff
@@ -1294,3 +1413,27 @@ func (dc *DeploymentController) rollbackToTemplate(deployment *extensions.Deploy
 	d, err = dc.updateDeploymentAndClearRollbackTo(deployment)
 	return
 }
+
+// isScalingEvent checks whether the provided deployment has been updated with a scaling event
+// by looking at the desired-replicas annotation in the active replica sets of the deployment.
+func (dc *DeploymentController) isScalingEvent(d *extensions.Deployment) bool {
+	newRS, oldRSs, err := dc.getAllReplicaSetsAndSyncRevision(d, false)
+	if err != nil {
+		return false
+	}
+	allRSs := append(oldRSs, newRS)
+	for _, rs := range controller.FilterActiveReplicaSets(allRSs) {
+		if desiredString, ok := rs.Annotations[deploymentutil.DesiredReplicasAnnotation]; ok {
```

Comment:
The rollout path will scale up only the new replica set, proportional scaling distributes for mitigating risk. This was asked actually by @bgrant0607. Also, in the future we will need to report progress for a rollout (progressDeadlineSeconds will keep track of intervals between scaling events happening in a rollout and fail a deployment that has no progress). Having external scaling events separate will help in that regard.

> the replicaCount and podTemplate maybe updated in a single etcd update

We can identify this in `isScalingEvent` and proceed with the rollout (because we need to rollout and not simply scale).

>Or a second update to etcd could be made before the sync is triggered for the first update.

I think clients of a watch will receive events on sequence, otherwise watches are broken, which is a bigger issue.